File: share-code/bai-tap-lap-trinh/python/string/65.py
import os

file_path = "D:/Data/Doing/psd/hailan.doc"

file_name_extension = os.path.splitext(os.path.basename(file_path))

file_name       = file_name_extension[0]
file_extension  = file_name_extension[1][1:]

print("Name:", file_name)
print("Extension:", file_extension)

File: share-code/bai-tap-lap-trinh/cpp/07-chuoi/65.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {
    string str1 = "D:/Data/Doing/psd/hailan.doc";
    size_t lastSlash = str1.find_last_of("/\\"); 
    
    if (lastSlash == string::npos) {
        cout << "Invalid file path." << endl;
    }
    
  
    string fileName = str1.substr(lastSlash + 1);
    size_t lastDot = fileName.find_last_of(".");
    if (lastDot == string::npos || lastDot == 0 || lastDot == fileName.length() - 1) {
        cout << "Invalid file extension." << endl;
    }
    string fileExtension = fileName.substr(lastDot + 1);
    string fileNameOnly = fileName.substr(0, lastDot);
    
   cout << "Name: " << fileNameOnly << endl;
    cout << "Extension: " << fileExtension << endl;

    return 0;
}